src/DataFilterer.ts

```diff
@@ -234,11 +234,13 @@ export class DataFilterer {
             // this is taken from https://stackoverflow.com/a/49717936
             const groupedSummed = new Map<string, number>();
             impactData.map((row) => {
-                 groupedSummed.set(row[compare], (groupedSummed.get(row[compare]) || 0) + row[metric]);
+                 groupedSummed.set(row[compare], 
```

Comment:
Seems like this is logic that would be good to get under test. I just ran `npm test`  to see what the state of tests were at the moment and it gave a few errors! It would be great to build up some test coverage and get this repo on travis to avoid regressions.

Reply:
I'll fix the broken test + add tests for this functionality on this branch.
It looks like I broke the tests by being overzealous in making methods private. Is there some best practice for the trade off between making member variable and methods private, and making them testable?

I'll create a ticket(s) for add more tests and then look at setting up code coverage.

Reply:
In the case of the `Filter.rangeValues` property, although it's not called by your code it is actually accessed by other code - it's bound to the view with knockout - so it makes sense for it to be public.

I can't think of a really succinct way to explain when to use private/public methods, but if you aren't familiar with SOLID principles of object oriented programming you might be interested to have a read: https://en.wikipedia.org/wiki/SOLID (personally I think S, I and D are the key ones.)

Basically private vals/methods relate to implementation details that other classes shouldn't depend on, whereas anything that is exposed to other classes via a public interface should be tested. E.g.

```
class SomeClass() {
   // other classes call me
   public function doSomething() {
      sharedCode();
      ... etc
   }

   // other classes call me 
   public function doSomethingElse() {
       sharedCode();
       ... etc
   }

   // implementation detail
   private function sharedCode() {
   }
}
``` 

If you do find yourself wanting to test a private method that's generally an indication that it doesn't belong in that class, so maybe what you want is something more like:

```
class SomeClass(private val helperClass) {

   public function doSomething() {
      helperClass.sharedCode();
      ... etc
   }

   public function doSomethingElse() {
       helperClass.sharedCode();
       ... etc
   }
}

class HelperClass() {
   function sharedCode() {
   }
}
``` 

It's all very subjective and more of an art than a science, hope this is at least somewhat helpful! Only really makes sense of course in the context of OO, with functional programming it's probably appropriate to export/test every method.